Getting-Started/Code/Debugging/Logging/index-v8.md

```diff
@@ -0,0 +1,250 @@
+
+
+```
+2018-11-12 08:34:50,419 [P27004/D2/T1] INFO   Umbraco.Core.Runtime.CoreRuntime - Booted. (4586ms) [Timing 9e76e5f]
+```
+
+Here is an example of the same log message represented as JSON, you can see here we have much more information that would allow us to search & filter logs based on these properties with an appropiate logging system.
+
+```json
+{
+  "@t": "2018-11-12T08:34:50.4190399Z",
+  "@mt": "{EndMessage} ({Duration}ms) [Timing {TimingId}]",
+  "EndMessage": "Booted.",
+  "Duration": 4586,
+  "TimingId": "9e76e5f",
+  "SourceContext": "Umbraco.Core.Runtime.CoreRuntime",
+  "ProcessId": 27004,
+  "ProcessName": "iisexpress",
+  "ThreadId": 1,
+  "AppDomainId": 2,
+  "AppDomainAppId": "LMW3SVC2ROOT",
+  "MachineName": "DELLBOOK",
+  "Log4NetLevel": "INFO ",
+  "HttpRequestNumber": 1,
+  "HttpRequestId": "557f45ba-0888-4216-8723-e226d795a2f7"
+}
+```
+
+To learn more about structured logging and message templates you can read more about it over on the https://messagetemplates.org website or alternatively watch this video from Serilog creator - https://www.youtube.com/watch?v=OhmNp8UPEEg
+
+## Writing to the log
+Umbraco writes log messages, but you are also able to use the Umbraco logger to write the log file as needed, so you can get further insights and details about your implementation.
+
+Here is a simple example of using the logger to write an Information message to the log which will contain one property of **Name** which will output the name variable that is passed into the method
+
+```csharp
+using Umbraco.Web.WebApi;
+using Umbraco.Core.Logging;
+
+namespace MyNamespace
+{
+    public class MyApiController : UmbracoApiController
+    {
+        public string GetSayHello(string name)
+        {
+            Logger.Info<MyApiController>("We are saying hello to {Name}", name);
+            return $"Hello {name}";
+        }
+    }
+}
+```
+
+The incorrect way to log the message would be use string interpolation or string concatanation such as
+
+```csharp
+//Do not use :(
+Logger.Info(typeof(MyApiController), $"We are saying hello to {name}");
+Logger.Info<MyApiController>($"We are saying hello to {name}");
+
+//Do not use :(
+Logger.Info(typeof(MyApiController), "We are saying hello to " + name);
+Logger.Info<MyApiController>("We are saying hello to " + name);
+```
+
+The above examples will write to the log file, however we will not get a seperate property logged with the message and we have no easy way to search for all log messages of this type.
+
+Where as the previous example we would be able to find all log messages that use the message template `We are saying hello to {Name}`
+
+If you are writing custom code and wish to use the logger, it's available in the following places:
+* SurfaceController
+* UmbracoApiController
+* UmbracoAuthorizedApiController
```

Comment:
not sure if this needs an example, it would be a repeat of the opening example I think, but maybe rewording this, just because it may not be clear 'how it's available' so more like...

If you are writing custom code and are inheriting from one of these special Umbraco base classes:

SurfaceController
UmbracoApiController
UmbracoAuthorizedApiController
RenderMvcController?

Then you can access the logging functionality via a special 'Logger' property included in those base classes (Do you need to add a using Umbraco.Core.Logging too?)

Outside of these places, eg a ContentFinder or your own custom code, you can get a reference to the logger from the Umbraco.Core.Composing namespace, using the 'Current.Logger' instance.

then I would put an example in of dong this...